independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/CreateProject.java

```diff
@@ -134,6 +147,15 @@ public boolean doCreateProject(final Map<String, Object> context) throws IOExcep
     }
 
     public QuarkusCommandOutcome execute() throws QuarkusCommandException {
+        // Define the Java version to use determined from the one specified or the one creating the project
+        Matcher matcher = JAVA_VERSION_PATTERN
+                .matcher(this.javaTarget != null ? this.javaTarget : System.getProperty("java.version", ""));
+        if (matcher.matches() && Integer.parseInt(matcher.group(1)) < 11) {
+            invocation.setProperty(JAVA_TARGET, invocation.getBuildTool() == BuildTool.MAVEN ? "1.8" : "1_8");
```

Comment:
why not have one "java_target" and another "java_underscore_target" so the same property isn't to be treated differently ?

Reply:
That was the other option I pondered. I decided to go with this one.